tests/test_halo_data_transformer.py

```diff
@@ -0,0 +1,452 @@
+    outside of it."""
+    sz = _shape_length(shape)
+    print(f"{shape} {sz}")
+    data = numpy.arange(0, sz, dtype=dtype).reshape(shape)
+    if backend == "gt4py_cupy":
+        quantity = Quantity(
+            cp.asnumpy(data),
+            dims=dims,
+            units=units,
+            origin=origin,
+            extent=extent,
+            gt4py_backend="gtcuda",
+        )
+        layout_map = quantity.storage.layout_map
+        print(layout_map)
+    elif backend == "gt4py_numpy":
+        quantity = Quantity(
+            data,
+            dims=dims,
+            units=units,
+            origin=origin,
+            extent=extent,
+            gt4py_backend="gtx86",
+        )
+        layout_map = quantity.storage.layout_map
+        print(layout_map)
+    else:
+        quantity = Quantity(data, dims=dims, units=units, origin=origin, extent=extent)
+    return quantity
+
+
+@pytest.fixture(params=[-0, -1, -2, -3])
+def rotation(request):
+    return request.param
+
+
+def test_data_transformer_allocate(quantity, n_halos):
+    data_transformer = HaloDataTransformer.get_from_numpy_module(quantity.np)
+    boundary_north = _boundary_utils.get_boundary_slice(
+        quantity.dims,
+        quantity.origin,
+        quantity.extent,
+        quantity.data.shape,
+        NORTH,
+        n_halos,
+        interior=False,
+    )
+    boundary_southwest = _boundary_utils.get_boundary_slice(
+        quantity.dims,
+        quantity.origin,
+        quantity.extent,
+        quantity.data.shape,
+        SOUTHWEST,
+        n_halos,
+        interior=False,
+    )
+
+    specification = HaloUpdateSpec(
+        shape=quantity.data.shape,
+        strides=quantity.data.strides,
+        itemsize=quantity.data.itemsize,
+        origin=quantity.metadata.origin,
+        extent=quantity.metadata.extent,
+        dims=quantity.metadata.dims,
+        numpy_module=quantity.np,
+        dtype=quantity.metadata.dtype,
+    )
+
+    data_transformer.add_scalar_specification(
+        specification, boundary_north, 0, boundary_north
+    )
+    data_transformer.add_scalar_specification(
+        specification, boundary_southwest, 0, boundary_southwest
+    )
+    data_transformer.compile()
+    assert len(data_transformer.get_send_buffer().array.shape) == 1
+    assert (
+        data_transformer.get_send_buffer().array.size
+        == quantity.data[boundary_north].size + quantity.data[boundary_southwest].size
+    )
+    assert len(data_transformer.get_recv_buffer().array.shape) == 1
+    assert (
+        data_transformer.get_recv_buffer().array.size
+        == quantity.data[boundary_north].size + quantity.data[boundary_southwest].size
+    )
+    # clean up
+    Buffer.push_to_cache(data_transformer._send_buffer)
+    Buffer.push_to_cache(data_transformer._recv_buffer)
+
+
+def _get_boundaries(quantity, n_halos):
+    boundary_send_interior = True
+    boundary_recv_interior = False
```

Comment:
Would these ever change, and do these variable names add context? I think the answer is no and no, which would mean you could hard-code these below.

Reply:
Removed